### PR TITLE
Support application key configuration

### PR DESCRIPTION
**Pull request by SaharBracha**

### CODE REVIEW

**Comment by SaharBracha** [comment]
recheck

**Comment by SaharBracha** [comment]
I have read the CLA Document and I hereby sign the CLA

File: lib/utils.js [hunk 1]
```diff
@@ -85,6 +87,65 @@ class Utils {
...
+        return __awaiter(this, void 0, void 0, function* () {
+            const appKey = core.getInput(Utils.APPLICATION_KEY);
+            if (appKey) {
+                return appKey;
+            }
+            const configFilePath = path.join(this.JFROG_CONFIG_DIR_NAME, this.JFROG_CONFIG_FILE);
+            try {
+                const config = yield this.readConfigFromFileSystem(configFilePath);
+                if (!config) {
+                    console.log('Config file is empty or not found.');
```
Comment (eyalbe4) [inline lib/utils.js:110]: console.log --> console.debug

File: lib/utils.js [hunk 2]
```diff
@@ -85,6 +87,65 @@ class Utils {
...
+                    return '';
+                }
+                const configObj = YAML.parse(config);
+                const applicationKey = configObj['application-key'];
+                if (!applicationKey) {
+                    console.log('Application key is not found in the config file.');
+                    return '';
+                }
+                console.log('Found application key: ' + applicationKey);
```
Comment (eyalbe4) [inline lib/utils.js:119]: console.log --> console.debug

File: lib/utils.js [hunk 3]
```diff
@@ -85,6 +87,65 @@ class Utils {
...
+                return applicationKey;
+            }
+            catch (error) {
+                console.log('Error reading config:', error);
```
Comment (eyalbe4) [inline lib/utils.js:123]: console.log --> console.error

File: src/utils.ts [hunk 4]
```diff
@@ -109,6 +123,70 @@ export class Utils {
...
+     * Retrieves the application key from input or .jfrog configuration file.
+     *
+     * This method attempts to retrieve the application key from the GitHub Action input.
+     * If the input is not provided, it reads .jfrog configuration file from the file system.
+     * If the configuration file exists and contains the application key, it returns the key.
+     * If the configuration file does not exist or does not contain the application key, it returns an empty string.
+     *
+     * @returns A promise that resolves to the application key as a string.
+     */
+    private static async getApplicationKey(): Promise<string> {
```
Comment (eyalbe4) [inline src/utils.ts:136]: Let's add a unit test for this function.
See the existing tests as a reference for the testing conventions.

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

File: src/utils.ts [hunk 5]
```diff
@@ -109,6 +123,70 @@ export class Utils {
         }
     }
 
+    /**
+     * Retrieves the application key from input or .jfrog configuration file.
```
[the remark this answers is not in the document]
Comment (SaharBracha) [reply src/utils.ts:127]: It's a requirement we got to support both of the options, we can discuss it

[the remark this answers is not in the document]
Comment (eyalbe4) [reply src/utils.ts:127]: Let's discuss this requirement, because the use case is unclear to me.

File: action.yml [hunk 6]
```diff
@@ -26,6 +26,9 @@ inputs:
     custom-server-id:
         description: "Custom JFrog CLI configuration server ID to use instead of the default one generated by the action."
         required: false
+    application-key:
+        description: "Application key to associate packages with."
+        required: false
```
Comment (eyalbe4) [inline action.yml:31]: Why is this input needed?

File: src/utils.ts [hunk 7]
```diff
@@ -109,6 +121,65 @@ export class Utils {
...
+     *
+     * This method attempts to retrieve the application key from the GitHub Action input.
+     * If the input is not provided, it reads .jfrog configuration file from the file system.
+     * If the configuration file exists and contains the application key, it returns the key.
+     * If the configuration file does not exist or does not contain the application key, it returns an empty string.
```
Comment (eyalbe4) [inline src/utils.ts:130]: This description is wrong, because the function should only read the application key from the confab file.

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

**Pull request merged**